### ISSUE TITLE
Expose HttpRequestMessage object to PowerShell function 

### ISSUE THREAD

**Issue by hebinhuang** [Azure/azure-webjobs-sdk-script#1171]
In C# HTTP trigger, the dev can use req (HttpRequestMessage object) to handle HTTP request - e.g. to get client certificate by calling req.GetClientCertificate(). 
But for PowerShell HTTP trigger, the req is an string object, there is no way for dev to deal with HTTP request message.

**Comment by mathewc** [comment]
While you are right that all aspects of the HttpRequestMessage may not be available, many of the important request details are available as environment variables, including:

* http method via REQ_METHOD
* headers via REQ_HEADERS_XXX
* query params via REQ_QUERY_XXX
* route params via REQ_PARAMS_XXX

**Comment by hebinhuang** [comment]
Correct, we have requirements to retrieve the client certificate from the HttpRequestsMessage. But, the PowerShell script hides the object of HttpRequestsMessage, so that there is no way to get the client certificate.

**Comment by hebinhuang** [comment]
@mathewc , any updates on this?

**Comment by mathewc** [comment]
Addressing this won't be a quick fix for us unfortunately. The issue is that certificates and other "advanced" request properties aren't easily mappable to environment vars like all the other request properties above. We can't currently just make the entire .NET HttpRequestMessage available to your code - that's not the current execution model.

Can you provide an example of the Powershell code you'd want to be able to write? How would you access/use this cert?
